File: call-clj/src/main/java/capclug/demo/callclj/Ten.java
/*
 * $Id$
 * Copyright (C) 2010 ARIN
 */
package capclug.demo.callclj;

import capclug.demo.cljfromjava.ten;

/**
 * @version $Rev$
 */
public class Ten
{
    public static void main( String[] args )
    {
        ten ten = new ten();
        ten.foo( "42d" );
        ten.bar( "43d" );
        ten.baz( "44d" ); // runtime exception!
    }
}

File: compile-clj-to-java/src/main/clojure/capclug/demo/cljfromjava/ten.clj
(ns capclug.demo.cljfromjava.ten
  (:gen-class
    :methods [[foo [String] void]
              [bar [String] void]
              [baz [String] void]]))

(defn -foo [this #^Integer baz]
  (println "will I work? " baz))

(defn -bar [this #^Integer baz]
  (println "will I work? " (.toUpperCase baz)))

(defn -baz [this #^Integer baz]
  (println "will I work? " (.doubleValue baz)))